### DISCUSSION TITLE
GPT-4-Vision gives wrong model name

### DISCUSSION THREAD

**Discussion by s2terminal** [danny-avila/LibreChat#2208] [Q&A]
### What happened?

Select a model such as GPT-3.5 in ChatGPT and attach an image. The screen display remains GPT-3.5, but the GPT-4-Vision is actually used.

### Steps to Reproduce

1. start LibreChat
1. select OpenAI gpt-3.5-turbo-1106
1. attach an image
1. send message
1. the reply will show GPT-3.5, but actually gpt-4-vision-preview should be running

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
2024-03-26 02:57:59 debug: [BaseClient] userMessage
{
  messageId: "bdfdcd52-0bcd-4129-8986-f79f6d187ed5",
  parentMessageId: "00000000-0000-0000-0000-000000000000",
  conversationId: "ca29c70b-73b3-4820-a1df-61a127aea7f6",
  sender: "User",
  text: "What is this?",
  isCreatedByUser: true,
  // 1 image_url(s)
  image_urls: [{"type":"image_url","image_url":{"url":"data:image/webp;base64,UklGRjZZAABXRUJQVlA4ICpZAABQwgGdASqAA... [truncated]],
  tokenCount: 773,
}
2024-03-26 02:57:59 debug: [OpenAIClient] chatCompletion
{
  baseURL: "https://api.openai.com/v1",
    modelOptions.model: "gpt-4-vision-preview",
    modelOptions.temperature: 1,
    modelOptions.top_p: 1,
    modelOptions.presence_penalty: 0,
    modelOptions.frequency_penalty: 0,
    modelOptions.user: "6601044fa58ec3190ee77749",
    modelOptions.stream: true,
    // 1 message(s)
    modelOptions.messages: [{"role":"user","content":[{"type":"text","text":"What is this?"},{"type":"image_url","image_url":{"u... [truncated]],
}
```

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/7953751/112b06f4-c8df-4641-81cf-a73368b00b8c)

(Image by <a href="https://pixabay.com/users/pexels-2286921/?utm_source=link-attribution&utm_medium=referral&utm_campaign=image&utm_content=1834639">Pexels</a> from <a href="https://pixabay.com//?utm_source=link-attribution&utm_medium=referral&utm_campaign=image&utm_content=1834639">Pixabay</a>)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This is not a bug, this is a feature. GPT-3.5 does not support images. Instead of giving you an error, the model is switched on your behalf.

If you want to remove this behavior, you need to remove `gpt-4-vision-preview` from your model list manually by using the `OPENAI_MODELS` environment variable to customize the list: https://docs.librechat.ai/install/configuration/dotenv.html#openai

**Reply by s2terminal** [reply]
>GPT-3.5 does not support images.

That is correct. But the screen shows that the image was read by GPT-3.5. The log shows that GPT-4-Vision is used. I consider this is a bug since the model actually used is different from the model displayed.

>If you want to remove this behavior, you need to remove gpt-4-vision-preview from your model list manually by using the OPENAI_MODELS environment variable to customize the list: https://docs.librechat.ai/install/configuration/dotenv.html#openai

On a different note than the issue, I removed gpt-4-vision from the environment variable OPENAI_MODELS, but when I uploaded the file, it still used gpt-4-vision. This is not a serious problem as I prefer to use gpt-4-vision.
